Code/Editor/EditorPreferencesDialog.cpp

```diff
@@ -309,30 +317,19 @@ void EditorPreferencesDialog::CreateImages()
 
 void EditorPreferencesDialog::CreatePages()
 {
-    std::vector<IClassDesc*> classes;
-    GetIEditor()->GetClassFactory()->GetClassesBySystemID(ESYSTEM_CLASS_PREFERENCE_PAGE, classes);
 
-    for (int i = 0; i < classes.size(); i++)
+    auto addPreferencePage = [&](IPreferencesPage* page) {
+       ui->pageTree->addTopLevelItem(new EditorPreferencesTreeWidgetItem(page, page->GetIcon()));
+    };
+    addPreferencePage(new CEditorPreferencesPage_General());
+    addPreferencePage(new CEditorPreferencesPage_Files());
+    addPreferencePage(new CEditorPreferencesPage_ViewportGeneral());
+    addPreferencePage(new CEditorPreferencesPage_ViewportCamera());
+    addPreferencePage(new CEditorPreferencesPage_ViewportManipulator());
+    addPreferencePage(new CEditorPreferencesPage_ViewportDebug());
+    if (AzToolsFramework::IsComponentWithServiceRegistered(AZ_CRC_CE("AWSCoreEditorService")))
     {
-        auto pUnknown = classes[i];
-
-        IPreferencesPageCreator* pPageCreator = nullptr;
-        if (FAILED(pUnknown->QueryInterface(&pPageCreator)))
-        {
-            continue;
-        }
-
-        int numPages = pPageCreator->GetPagesCount();
-        for (int pindex = 0; pindex < numPages; pindex++)
-        {
-            IPreferencesPage* pPage = pPageCreator->CreateEditorPreferencesPage(pindex);
-            if (!pPage)
-            {
-                continue;
-            }
-
-            ui->pageTree->addTopLevelItem(new EditorPreferencesTreeWidgetItem(pPage, pPage->GetIcon()));
-        }
+        addPreferencePage(new CEditorPreferencesPage_AWS());
```

Comment:
As much as I don't like this, its similar to the existing code, that is, `CEditorPreferencesPage_AWS` is a type inside Code/Editor, not from the gem (which it should be, but isn't).

I can't fault this since its just moving it from PreferencesStdPages (in Code/Editor) to EditorPreferencesDlg (in Code/Editor) and we don't generally ask for a lot of new code during dead code removal or refactor.

If I had known about this when it was initially written I would have indeed asked for a registration EBUS and a proper API for adding preference pages.  we do need that, but we dont' need that as part of a dead-code refactor.

Reply:
If anything, we should consider adding a new feature issue or ticket to establish an ebus for this, probably just a simple request response one, where, for example, there's an ebus that "things that want to talk to the Preferences system" can listen for events on (one-to-many bus/interface) which contains a "Register Preference Page" callback fn (or "factory a preference page" function to return the actual widget or whatever).  As long as there's a Qt base class thats DLLExported somewhere to contian the base preference page and its necessary signals and slots it should work fine as the API itself for eveyrthing else.

Reply:
here you go: https://github.com/o3de/o3de/issues/18074